packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin.swift

```diff
@@ -1,19 +1,111 @@
+        let pluginInstance = AmplifyPushNotificationsPlugin(
+            sharedEventsStreamHandlers: EventsStreamHandlers(binaryMessenger: messenger)
+        )
+
+        NativePushNotificationsPluginSetup(messenger, pluginInstance)
+
+        registrar.addApplicationDelegate(pluginInstance)
+    }
+
+    public func getPermissionStatus(completion: @escaping (String?, FlutterError?) -> Void) {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            var status: String
+
+            switch settings.authorizationStatus {
+            case .notDetermined:
+                status = "NotDetermined"
+            case .denied:
+                status = "Denied"
+            case .authorized:
+                status = "Authorized"
+            case .ephemeral:
+                status = "Ephemeral"
+            case .provisional:
+                status = "Provisional"
+            @unknown default:
+                status = "NotDetermined"
+            }
+
+            completion(status, nil)
+        }
+    }
+
+
+    public func requestPermissions(withPermissionOptions: PermissionsOptions, completion: @escaping (NSNumber?, FlutterError?) -> Void) {
+        var options: UNAuthorizationOptions = []
+
+        if withPermissionOptions.alert.boolValue {
+            options.insert(.alert)
+        }
+
+        if withPermissionOptions.badge.boolValue {
+            options.insert(.badge)
+        }
+
+        if withPermissionOptions.sound.boolValue {
+            options.insert(.sound)
+        }
+
+        UNUserNotificationCenter.current().requestAuthorization(options: options) { granted, error in
+            if let error = error {
+                completion(
+                    nil,
+                    FlutterError(
+                        code: "RequsetPermissionsError",
+                        message: "Error occurred requesting notitication center authorization.",
+                        details: error.localizedDescription
+                    )
+                )
+            } else {
+                completion(granted as NSNumber, nil)
+            }
+        }
+    }
+
+    public func getLaunchNotificationWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>) -> [AnyHashable : Any]? {
+        let launchNotification = self.launchNotification
+        self.launchNotification = nil
+        return launchNotification
+    }
+
+
+    public func getBadgeCountWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>) -> NSNumber? {
+        return UIApplication.shared.applicationIconBadgeNumber as NSNumber
+    }
+
+    public func setBadgeCountWithBadgeCount(_ withBadgeCount: NSNumber, error: AutoreleasingUnsafeMutablePointer<FlutterError?>) {
+        UIApplication.shared.applicationIconBadgeNumber = withBadgeCount.intValue
+    }
+
+    public func completeNotification(withCompletionHandlerId: String, error: AutoreleasingUnsafeMutablePointer<FlutterError?>) {
```

Comment:
Was thinking about doing the way you described. I didn't do so because this setup is pretty much iOS specific - it runs the callback function after the App wakes up, where Android creates a "headless" Dart isolate to run the callback, so Android unnecessarily need this platform channel call into Flutter. Will update.

Reply:
Updated 👍